qml/keys/CharKey.qml

```diff
@@ -101,12 +116,16 @@ Item {
 
     Text {
         id: keyLabel
-        text: (panel.activeKeypadState === "NORMAL") ? label : shifted;
-        anchors.centerIn: parent
+        text: (imgNormal === UI.imageGreyKey[formFactor]) ? label : (panel.activeKeypadState === "NORMAL") ? label : shifted;
+        anchors.horizontalCenter: buttonImage.horizontalCenter
+		anchors.verticalCenter: buttonImage.verticalCenter 
+		anchors.verticalCenterOffset: imgNormal === UI.imageGreyKey[formFactor] ? units.gu(0.5) : units.gu(-0.25)
         font.family: UI.fontFamily
-        font.pixelSize: fontSize
+        font.pixelSize: ((imgNormal === UI.imageGreyKey[formFactor])&&(panel.activeKeypadState !== "NORMAL") ) ? UI.annotationFontSize : fontSize
         font.bold: UI.fontBold
-        color: UI.fontColor
+        color: ((imgNormal === UI.imageGreyKey[formFactor])&&(panel.activeKeypadState !== "NORMAL")) ? UI.annotationFontColor : UI.fontColor
```

Comment:
same question about the logic; I feel we could have an additional parameter in CharKey that would distinguish the annotation case from the normal case